packages/time-series/lib/commands/index.ts

```diff
@@ -137,28 +141,75 @@ export function pushLabelsArgument(args: Array<RedisArgument>, labels?: Labels)
   return args;
 }
 
-export type SampleRawReply = {
-  2: TuplesReply<[timestamp: NumberReply, value: BlobStringReply]>;
-  3: TuplesReply<[timestamp: NumberReply, value: DoubleReply]>;
-};
+export type SampleRawReply = TuplesReply<[timestamp: NumberReply, value: DoubleReply]>;
 
 export const transformSampleReply = {
-  2(reply: SampleRawReply[2]) {
-    const [timestamp, value] = reply as unknown as UnwrapReply<typeof reply>;
+  2(reply: Resp2Reply<SampleRawReply>) {
+    const [ timestamp, value ] = reply as unknown as UnwrapReply<typeof reply>;
     return {
       timestamp,
       value: Number(value)
     };
   },
-  3(reply: SampleRawReply[3]) {
-    const [timestamp, value] = reply as unknown as UnwrapReply<typeof reply>;
+  3(reply: SampleRawReply) {
+    const [ timestamp, value ] = reply as unknown as UnwrapReply<typeof reply>;
     return {
       timestamp,
       value
     };
   }
 };
 
+export type SamplesRawReply = ArrayReply<SampleRawReply>;
+
+export const transformSamplesReply = {
+  2(reply: Resp2Reply<SamplesRawReply>) {
+    return (reply as unknown as UnwrapReply<typeof reply>)
+      .map(sample => transformSampleReply[2](sample));
+  },
+  3(reply: SamplesRawReply) {
+    return (reply as unknown as UnwrapReply<typeof reply>)
+      .map(sample => transformSampleReply[3](sample));  }
+};
+
+export function transformLablesReply2(r: RawLabels2): Labels {
+  const labels: Labels = {};
+
+  const reply = r as unknown as UnwrapReply<typeof r>;
+
+  for (const t of reply) {
+    const [k, v] = t as unknown as UnwrapReply<typeof t>;
+    const key = k as unknown as UnwrapReply<BlobStringReply>;
+    const value = v as unknown as UnwrapReply<BlobStringReply>;
+
+    labels[key.toString()] = value.toString()
+  }
+
+  return labels
+}
+
+export function transformLablesReply3(r: RawLabels3): Labels {
+  const reply = r as unknown as UnwrapReply<RawLabels3>;
```

Comment:
why labels is always an object and not "MapReply"?

Reply:
So, the Q is, what does returning it as all possible map reply type mappings give us (over a simple object).  i.e. related to above Q.  If we define that we only support labels as string->string, I'd argue it gives us minimal if anything over a simple object.  If we want to support binary labels and binary values, then we do gain something.